### PR TITLE
7 complete

### PR DESCRIPTION
**Pull request by everydaycloud**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -219,3 +219,61 @@ describe("GET /api/articles/:article_id/comments get comments by article id", ()
...
+        .then((res)=>{
+            expect(res.body).hasOwnProperty("author")
+            expect(res.body).hasOwnProperty("body")
+            expect(res.body).hasOwnProperty("article_id")
+            expect(res.body).hasOwnProperty("comment_id")
+            expect(res.body).hasOwnProperty("votes")
+            expect(res.body).hasOwnProperty("created_at")
+        })
+    });
+    test('responds with status 400 error when request is incomplete', () => {
```
Comment (Beth-McCormack) [inline __tests__/app.test.js:254]: A great test to include👌

File: __tests__/app.test.js [hunk 2]
```diff
@@ -219,3 +219,61 @@ describe("GET /api/articles/:article_id/comments get comments by article id", ()
...
+    test("should return a 404 status and message 'article not found' if the article does not exist", () => {
+        const addedComment = {"username":"poke", "body":"mon"};
+        return request(app)
+          .post("/api/articles/9999/comments")
+          .send(addedComment)
+          .expect(404)
+          .then((res) => {
+            expect(res.body.msg).toBe("Article not found")
+          });
+      });
```
Comment (Beth-McCormack) [inline __tests__/app.test.js:293]: This is really thorough testing, well done! I would consider adding a test for if the comment has extra, unnecessary properties, eg 
{ 
    username: "butter_bridge"
    body: "comment"
    date: 1/1/23
}

File: __tests__/app.test.js [hunk 3]
```diff
@@ -219,3 +219,61 @@ describe("GET /api/articles/:article_id/comments get comments by article id", ()
...
+        return request(app)
+          .post("/api/articles/3/comments")
+          .send(addedComment)
+          .expect(404)
+          .then((res) => {
+            expect(res.body.msg).toBe("Author not found");
+          });
+      });
+    test("should return a 404 status and message 'article not found' if the article does not exist", () => {
+        const addedComment = {"username":"poke", "body":"mon"};
```
Comment (Beth-McCormack) [inline __tests__/app.test.js]: For these tests, consider using a username which already exists to ensure the error being returned is for the right reasons

**Review by Beth-McCormack — COMMENTED** [review]
Feel free to merge once you've had a look at my comments. Well done!

File: __tests__/app.test.js (hunk 3 above)
Comment (everydaycloud) [reply __tests__/app.test.js]: all done, thanks Beth!

**Pull request merged**